docs/how-to/write-tests.md

```diff
@@ -6,3 +6,130 @@ Testing is essential to maintain the integrity and reliability of the codebase.
+```
+
+In this example, we need to test the `stage`, `unstage`, and `read` methods. The `read` method needs to be tested because we are adding `signal_a` and `signal_b` as readables to the device, so we need to confirm that we get the expected read signals back when this method is called.
+
+We use [pytest](https://docs.pytest.org/en/stable/contents.html) for writing tests in dodal. A core part of this library is the use of fixtures. A fixture is a function decorated with `@pytest.fixture` that provides setup/teardown or reusable test data for your tests. It is defined once and can be reused across multiple tests. Fixtures are mainly used to define devices and then inject them into each test.
+
+To help set up a device, they are created with the `init_devices(mock=True)` function from `ophyd_async.core`, which automatically initialises the device in mock mode. The `RunEngine` is present when creating a device to ensure there is an event loop available when connecting signals. The `RunEngine` fixture itself is defined in `dodal/tests/conftest.py`, which is a special file recognised by `pytest`. It defines fixtures that are automatically available to any test in the same directory (and its subdirectories) without needing to import them. This is useful for defining common setup code once without duplicating it across test files.
+
+In order for `pytest` to detect something as a test, a function should begin with `test_*`. The test function should be self-descriptive about what it is testing, and it is acceptable (even encouraged) to have longer names for test functions for clarity.
+
+```Python
+import asyncio
+from unittest.mock import ANY
+
+import pytest
+from bluesky import RunEngine
+from bluesky import plan_stubs as bps
+from ophyd_async.core import OnOff, init_devices
+from ophyd_async.testing import assert_reading, get_mock_put, partial_reading
+from dodal.device.my_device import  MyDevice
+
+
+# RunEngine is needed to make sure there is an event loop when creating device.
+@pytest.fixture
+async def sim_my_device(RE: RunEngine) -> MyDevice:
+    async with init_devices(mock=True):
+        sim_my_device = MyDevice("TEST:")
+    return sim_my_device
+
+
+def test_my_device_stage(sim_my_device: MyDevice, RE: RunEngine) -> None:
+    RE(bps.stage(sim_my_device, wait=True), wait=True)
+    get_mock_put(sim_my_device.signal_b).assert_called_once_with(OnOff.ON, wait=True)
+
+
+def test_my_device_unstage(sim_my_device: MyDevice, RE: RunEngine) -> None:
+    RE(bps.unstage(sim_my_device, wait=True), wait=True)
+    get_mock_put(sim_my_device.signal_b).assert_called_once_with(OnOff.OFF, wait=True)
+
+
+async def test_my_device_read(sim_my_device: MyDevice, RE: RunEngine) -> None:
+    prefix = sim_my_device.name
+    await assert_reading(
+        sim_my_device,
+        {
+            f"{prefix}-signal_a": partial_reading(ANY),
+            f"{prefix}-signal_b": partial_reading(ANY),
```

Comment:
```suggestion
            f"{prefix}-signal_a": partial_reading(0),
            f"{prefix}-signal_b": partial_reading(OnOff.OFF),
```

If you're going to test the read method, you must check a real expected value. As discussed above, I don't think it's necessary unless the reading behaviour is complex.